dashboard/app/serializers/api/v1/pd/workshop_enrollment_serializer.rb

```diff
@@ -11,12 +11,17 @@ def user_id
     user ? user.id : nil
   end
 
-  def alternate_email
+  def application_id
     application_id = object.try(:application_id)
+    return unless application_id && Pd::Application::TeacherApplication.exists?(id: application_id)
+    application_id
+  end
+
+  def alternate_email
     return unless application_id
 
     # Note: Use dig instead of [] because RuboCop doesn't like chaining ordinary method call after safe navigation operator.
-    Pd::Application::ApplicationBase.find(application_id)&.
+    Pd::Application::TeacherApplication.find(application_id)&.
```

Comment:
We should only be linking teacher applications here –– principal approval applications shouldn't be linked, and no other applications are currently being used.

Reply:
I think you could use `object&.application` in place of lines 21 and 24 too